build failure with openjdk-17

Using latest Debian 11 java17 :
```
java --version                                                                                                                                                 leshan/git/master 
openjdk 17.0.4 2022-07-19
OpenJDK Runtime Environment (build 17.0.4+8-Debian-1deb11u1)
OpenJDK 64-Bit Server VM (build 17.0.4+8-Debian-1deb11u1, mixed mode, sharing)
```
I have some tests falling, example:
```
[ERROR] Failures: 
[ERROR]   LeshanServerBuilderTest.create_server_without_psk_cipher:134
[ERROR]   LeshanBootstrapServerBuilderTest.create_server_without_psk_cipher:207
```

Logs for LeshanBootstrapServerBuilderTest.create_server_without_psk_cipher
```
11:17:45.739 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/0-1_1.xml
11:17:45.855 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/0.xml
11:17:46.033 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/1-1_0.xml
11:17:46.118 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/1-1_1.xml
11:17:46.274 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/2-1_0.xml
11:17:46.338 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/2.xml
11:17:46.403 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/3-1_0.xml
11:17:46.552 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/3-1_1.xml
11:17:46.694 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/3.xml
11:17:46.831 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/4-1_0.xml
11:17:46.921 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/4-1_1.xml
11:17:47.022 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/4-1_2.xml
11:17:47.121 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/4.xml
11:17:47.226 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/5-1_0.xml
11:17:47.317 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/5.xml
11:17:47.430 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/6.xml
11:17:47.507 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/7.xml
11:17:47.591 [main] DEBUG org.eclipse.leshan.core.model.DDFFileParser - Parsing DDF file /models/21.xml
11:17:51.088 [main] WARN org.eclipse.leshan.server.californium.bootstrap.LeshanBootstrapServerBuilder - Unable to create DTLS config and so secured endpoint.
java.lang.IllegalStateException: EC key pair is not valid!
	at org.eclipse.californium.scandium.dtls.x509.SingleCertificateProvider.setupConfigurationHelper(SingleCertificateProvider.java:198)
SingleCertificateProvider.java:198
	at org.eclipse.californium.scandium.config.DtlsConnectorConfig$Builder.build(DtlsConnectorConfig.java:2324)
DtlsConnectorConfig.java:2324
	at org.eclipse.leshan.server.californium.bootstrap.LeshanBootstrapServerBuilder.build(LeshanBootstrapServerBuilder.java:557)
LeshanBootstrapServerBuilder.java:557
	at org.eclipse.leshan.server.californium.bootstrap.LeshanBootstrapServerBuilderTest.create_server_without_psk_cipher(LeshanBootstrapServerBuilderTest.java:206)
LeshanBootstrapServerBuilderTest.java:206
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
NativeMethodAccessorImpl.java:77
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
DelegatingMethodAccessorImpl.java:43
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Method.java:568
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
FrameworkMethod.java:59
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
ReflectiveCallable.java:12
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
FrameworkMethod.java:56
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
InvokeMethod.java:17
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
RunBefores.java:26
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
ParentRunner.java:306
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
BlockJUnit4ClassRunner.java:100
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
ParentRunner.java:366
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
BlockJUnit4ClassRunner.java:103
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
BlockJUnit4ClassRunner.java:63
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
ParentRunner.java:331
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
ParentRunner.java:79
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
ParentRunner.java:329
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
ParentRunner.java:66
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
ParentRunner.java:293
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
ParentRunner.java:306
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
ParentRunner.java:413
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
11:17:51.346 [main] INFO org.eclipse.californium.core.network.RandomTokenGenerator - using tokens of 8 bytes in length
11:17:51.772 [main] DEBUG org.eclipse.californium.core.network.InMemoryMessageExchangeStore - [LWM2M BS Server-coap://] using TokenProvider org.eclipse.californium.core.network.RandomTokenGenerator
11:17:52.149 [main] INFO org.eclipse.californium.ban - Started.
11:17:52.204 [main] INFO org.eclipse.californium.core.network.CoapEndpoint - [LWM2M BS Server-coap://] CoapEndpoint uses udp context
11:17:52.749 [main] INFO org.eclipse.californium.core.network.stack.BlockwiseLayer - [LWM2M BS Server-coap://] BlockwiseLayer uses MAX_MESSAGE_SIZE=1024, PREFERRED_BLOCK_SIZE=512, BLOCKWISE_STATUS_LIFETIME=300000, MAX_RESOURCE_BODY_SIZE=8192, BLOCKWISE_STRICT_BLOCK2_OPTION=false
```

--- a/leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/BootstrapIntegrationTestHelper.java
+++ b/leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/BootstrapIntegrationTestHelper.java
@@ -140,9 +140,9 @@ public class BootstrapIntegrationTestHelper extends SecureIntegrationTestHelper 
             ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
             // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
+            KeySpec publicKeySpec = new ECPublicKeySpec(
+                    new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)), parameterSpec);
+            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(1, privateS), parameterSpec);
 
             // Get keys
             bootstrapServerPublicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);

--- a/leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/SecureIntegrationTestHelper.java
+++ b/leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/util/SecureIntegrationTestHelper.java
@@ -165,9 +165,9 @@ public class SecureIntegrationTestHelper extends IntegrationTestHelper {
             ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
             // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
+            KeySpec publicKeySpec = new ECPublicKeySpec(
+                    new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)), parameterSpec);
+            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(1, privateS), parameterSpec);
 
             // Get keys
             clientPublicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);
@@ -208,9 +208,9 @@ public class SecureIntegrationTestHelper extends IntegrationTestHelper {
             ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
             // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
+            KeySpec publicKeySpec = new ECPublicKeySpec(
+                    new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)), parameterSpec);
+            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(1, privateS), parameterSpec);
 
             // Get keys
             serverPublicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);

--- a/leshan-server-cf/src/test/java/org/eclipse/leshan/server/californium/LeshanServerBuilderTest.java
+++ b/leshan-server-cf/src/test/java/org/eclipse/leshan/server/californium/LeshanServerBuilderTest.java
@@ -64,9 +64,9 @@ public class LeshanServerBuilderTest {
             ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
             // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
+            KeySpec publicKeySpec = new ECPublicKeySpec(
+                    new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)), parameterSpec);
+            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(1, privateS), parameterSpec);
 
             // Get keys
             publicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);

--- a/leshan-server-cf/src/test/java/org/eclipse/leshan/server/californium/bootstrap/LeshanBootstrapServerBuilderTest.java
+++ b/leshan-server-cf/src/test/java/org/eclipse/leshan/server/californium/bootstrap/LeshanBootstrapServerBuilderTest.java
@@ -73,9 +73,9 @@ public class LeshanBootstrapServerBuilderTest {
             ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
             // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
+            KeySpec publicKeySpec = new ECPublicKeySpec(
+                    new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)), parameterSpec);
+            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(1, privateS), parameterSpec);
 
             // Get keys
             publicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);

--- a/leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/servlet/json/JacksonSecurityDeserializer.java
+++ b/leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/servlet/json/JacksonSecurityDeserializer.java
@@ -92,7 +92,7 @@ public class JacksonSecurityDeserializer extends JsonDeserializer<SecurityInfo> 
                         algoParameters.init(new ECGenParameterSpec(params));
                         ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
-                        KeySpec keySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(x), new BigInteger(y)),
+                        KeySpec keySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(1, x), new BigInteger(1, y)),
                                 parameterSpec);
 
                         key = KeyFactory.getInstance("EC").generatePublic(keySpec);

--- a/leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/serialization/SecurityInfoSerDes.java
+++ b/leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/serialization/SecurityInfoSerDes.java
@@ -106,7 +106,8 @@ public class SecurityInfoSerDes {
                 algoParameters.init(new ECGenParameterSpec(params));
                 ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
-                KeySpec keySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(x), new BigInteger(y)), parameterSpec);
+                KeySpec keySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(1, x), new BigInteger(1, y)),
+                        parameterSpec);
 
                 key = KeyFactory.getInstance("EC").generatePublic(keySpec);
 

--- a/leshan-server-redis/src/test/java/org/eclipse/leshan/server/redis/serialization/SecurityInfoSerDesTest.java
+++ b/leshan-server-redis/src/test/java/org/eclipse/leshan/server/redis/serialization/SecurityInfoSerDesTest.java
@@ -55,7 +55,7 @@ public class SecurityInfoSerDesTest {
         ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
 
         // Create key specs
-        KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
+        KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(1, publicX), new BigInteger(1, publicY)),
                 parameterSpec);
 
         SecurityInfo si = SecurityInfo.newRawPublicKeyInfo("myendpoint",

